homeassistant/components/habitica/sensor.py

```diff
@@ -259,14 +261,102 @@ async def async_setup_entry(
     """Set up the habitica sensors."""
 
     coordinator = config_entry.runtime_data
+    ent_reg = er.async_get(hass)
+    entities: list[SensorEntity] = []
+    description: SensorEntityDescription
+    for description in SENSOR_DESCRIPTIONS:
+        if description.key is HabiticaSensorEntity.HEALTH_MAX:
+            if entity_id := ent_reg.async_get_entity_id(
```

Comment:
Maybe this whole code-block could go into a separate function and call the function. Then we can make the code even more dry by combining it with the task sensors also below.